src/rest/connector/libs/nso/implementation.py

```diff
@@ -129,7 +129,11 @@ def connect(self,
         self.session.auth = (username, password)
 
         # Connect to the device via requests
-        response = self.session.get(self.login_url, timeout=timeout)
+        for login_url in self.login_urls:
+            response = self.session.get(login_url, timeout=timeout)
+            if response.status_code == requests.codes.ok:
+                self.login_url = login_url
+                break
         output = response.text
         log.debug("Response: {c} {r}, headers: {h}".format(c=response.status_code,
             r=response.reason, h=response.headers))
```

Comment:
Please move the debug logging into the for loop

Reply:
Done